crates/hyperqueue/src/transfer/stream.rs

```diff
@@ -26,25 +25,27 @@ pub struct DataMsg {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct EndTaskStreamMsg {
+pub struct StreamTaskEnd {
+    pub job: JobId,
     pub task: JobTaskId,
     pub instance: InstanceId,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub enum FromStreamerMessage {
-    Start(StartTaskStreamMsg),
-    Data(DataMsg),
-    End(EndTaskStreamMsg),
-}
-
-#[derive(Serialize, Deserialize, Debug)]
-pub struct EndTaskStreamResponseMsg {
+pub struct StreamChunkHeader {
+    #[serde(with = "ts_milliseconds")]
+    pub time: DateTime<Utc>,
+    pub job: JobId,
     pub task: JobTaskId,
+    pub instance: InstanceId,
+    pub channel: ChannelId,
+    pub size: u64,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-pub enum ToStreamerMessage {
-    Error(String),
-    EndResponse(EndTaskStreamResponseMsg),
+pub enum StreamerMessage {
```

Comment:
This should probably be in `src/worker/streamer.rs`? It does not implement serde traits and it's currently only used in that file.